pontoon/sync/utils.py

```diff
@@ -132,6 +132,21 @@ def source_to_locale_path(path):
     return path
 
 
+def escape_apostrophes(value):
+    """
+    Apostrophes (straight single quotes) have special usage in Android strings.xml files,
+    so they need to be escaped according to the standard XML/HTML escaping rules.
```

Comment:
I see a lot of unescaped `?` in our files:
https://github.com/mozilla-l10n/android-l10n/blob/main/mozilla-mobile/fenix/app/src/main/res/values/strings.xml#L82C98-L82C99

I also don't remember any reports on `@` and `?` being an issue or anything related in https://github.com/mozilla-l10n/mozl10n-linter/. 

@flodolo Should we do anything special with these two characters as suggested in the Android docs? https://developer.android.com/guide/topics/resources/string-resource.html#escaping_quotes

Reply:
It's the first I hear about `?` and `@` requiring escape, so it's not surprising that the linter doesn't have anything about them.

I'm sure there are Android docs somewhere, but this is from [LingoHub](https://lingohub.com/developers/resource-files/android-localization)

> The characters “@” and “?” are escaped with “\” if they are at the beginning of the content (or represent the only content) unless the whole string is quoted using double quotes (“) as well or they represent references (e.g. in the form @string/keyName)

Reply:
In this case I'd stay away from escaping `?` and `@` for now, since this PR is a just a fix for https://github.com/mozilla/pontoon/issues/2212, and a temporary one.